sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_region.c

```diff
@@ -48,6 +60,19 @@ int gnrc_lorawan_set_dr(gnrc_lorawan_t *mac, uint8_t datarate)
     return 0;
 }
 
+int gnrc_lorawan_set_tx_power(gnrc_lorawan_t *mac, uint8_t tx_pwr)
+{
+    netdev_t *dev = gnrc_lorawan_get_netdev(mac);
+
+    if (!gnrc_lorawan_validate_tx_power(tx_pwr)) {
+        return -EINVAL;
+    }
+    DEBUG("gnrc_lorawan_region: TX Power index: %u \n",tx_pwr);
+
+    int tx_pwr_db = GNRC_LORAWAN_TX_POWER_MAX - (tx_pwr * 2);
```

Comment:
```suggestion
    uint16_t tx_pwr_db = GNRC_LORAWAN_TX_POWER_MAX - (tx_pwr * 2);
```

NETOPT_TX_POWER requires `int16_t`. This will probably crash on 32-bit platforms.